### DISCUSSION TITLE
Optimize JavaScript "above the fold" / Reduce unused javascript

### DISCUSSION THREAD

**Discussion by schaschko** [vikejs/vike#908] [Discussion]
A page I'm building has some potential in optimizing speed as lighthouse suggests:

![image](https://github.com/brillout/vite-plugin-ssr/assets/57093148/5b2fce2a-ba32-4580-9cf5-cdf4d704e115)

I searched through vps, vite and the web, and the issue seems to be about code splitting into above / below the fold. Are there any reasonable (automated) optimization techniques one can utilize with vps /vite tools etc. to tackle the problem?

What is best practice in such a case?

**Comment by brillout** [comment]
I stand to be corrected, but I don't think there is an (easy) way to automate this for JavaScript code &ndash; I think it needs to be done manually.

There is a library that does help regarding third-party libraries. (I don't recall its name, something with "party".)

**Reply by schaschko** [reply]
That's also my impression. Would be interested to hear what others think about it or how they did it. I'm certainly not the first one to encounter this problem.

Fiddling around with [chunking strategy](https://vitejs.dev/guide/build.html#chunking-strategy) did not help, at least in my case.

**Reply by spacedawwwg** [reply]
https://partytown.builder.io/

**Reply by brillout** [reply]
> https://partytown.builder.io/

👍 Thanks. That's the one I was thinking about.
